src/jira-client.js

```diff
@@ -1,89 +1,114 @@
+  return `${session.name}=${session.value}`;
+}
+
+function makeJiraUri({ baseUri = getCurrentContext().uri, uri, query } = {}) {
+  const fullUri = new URL(`/rest/${ uri }`, `${baseUri}`);
+  if (query) {
+    Object.entries(query).forEach(([key, value]) => {
+      fullUri.searchParams.set(key, value);
     });
-    return `${session.name}=${session.value}`;
-  } catch (err) {
-    throw new Error(err.error.errorMessages ? err.error.errorMessages.join(', ') : err.message);
   }
-}
 
-function makeJiraUri({ baseUri = getCurrentContext().uri, uri } = {}) {
-  return `${baseUri}/rest/${ uri }`;
+  const uriStr = format(fullUri);
+  debug.verbose('Make URI', uriStr);
+  return uriStr;
 }
 
 async function makeQuery({ jql, selector = (results) => results.total } = {}) {
-  const points = getCurrentContext().points;
-  try {
-    const response = await rp(Object.assign({}, await getRequestOptions(), {
-      method: 'POST',
-      uri: makeJiraUri({ uri: 'api/2/search' }),
-      body: {
-        jql,
-        startAt: 0,
-        maxResults: 10000,
-        fields: ['summary', 'status', 'assignee', 'description', points],
-        expand: ['schema', 'names']
-      }
-    }));
-    return selector(response);
-  } catch (err) {
-    throw new Error(err.error.errorMessages.join(', '));
-  }
+  const { points } = getCurrentContext();
+
+  const opts = await getRequestOptions();
+  const uri = makeJiraUri({ uri: 'api/2/search' });
+  debug.http(`POST ${uri}`);
+
+  // TODO: move to fetch from rp
+  const response = await rp(Object.assign({}, opts, {
+    method: 'POST',
+    uri,
+    body: {
+      jql,
+      startAt: 0,
+      maxResults: 10000,
+      fields: ['summary', 'status', 'assignee', 'description', points],
+      expand: ['schema', 'names']
+    }
+  }));
+
+  return selector(response);
 }
 
 async function makeGetRequest(url, api = 'agile/1.0', options = {}) {
-  try {
-    return await rp(Object.assign({}, await getRequestOptions(), options, {
-      method: 'GET',
-      uri: makeJiraUri({ uri: `${api}/${url}` })
-    }));
-  } catch (err) {
-    throw new Error(err.error.errorMessages.join(', '));
-  }
+  const opts = await getRequestOptions();
+  const uri = makeJiraUri({ uri: `${api}/${url}`, query: options.query });
+  const fullOpts = Object.assign({}, opts, options, {
+    method: 'GET',
+    uri
+  });
+
+  debug.http(`GET ${uri}`, fullOpts);
+  return await rp(fullOpts);
 }
 
 async function makePutRequest(url, api = 'agile/1.0', data = {}) {
-  try {
-    return await rp(Object.assign({}, await getRequestOptions(), {
-      method: 'PUT',
-      uri: makeJiraUri({ uri: `${api}/${url}` }),
-      body: data
-    }));
-  } catch (err) {
-    throw new Error(err.error.errorMessages.join(', '));
-  }
+  const opts = await getRequestOptions();
+  const uri = makeJiraUri({ uri: `${api}/${url}` });
+  debug.http(`PUT ${uri}`);
+
+  return await rp(Object.assign({}, opts, {
+    method: 'PUT',
+    uri,
+    body: data
+  }));
```

Comment:
Here's an example of what I was referring to:

```suggestion
  try {
    return await rp(Object.assign({}, opts, {
      method: 'PUT',
      uri,
      body: data
    }));
  } catch (err) {
    throw new Error(err.error.errorMessages.join(', '));
  }
```

Reply:
Ahh. That makes sense. Thank you!